### PR TITLE
transaction cost for the operator

### PR DESCRIPTION
**Pull request by sinev-valentine**
the logs of the proxy contains the cost of each transaction for the operator.

### CODE REVIEW

File: proxy/plugin/solana_rest_api_tools.py [hunk 1]
```diff
@@ -472,40 +472,40 @@ def check_if_continue_returned(result):
     return (False, ())
 
 
-def call_continue_0x0d(signer, client, perm_accs, trx_accs, steps, msg):
+def call_continue_0x0d(signer, client, perm_accs, trx_accs, steps, msg, ethTrx):
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: `ethTrx` in other places called `eth_trx`. It looks strange one thing has different names in one PR.

File: proxy/plugin/solana_rest_api_tools.py [hunk 2]
```diff
@@ -527,16 +527,21 @@ def call_continue_bucked(signer, client, perm_accs, trx_accs, steps):
...
             result = client.get_confirmed_transaction(trx)
+            update_transaction_cost(result, ethTrx, extra_sol_trx=extra_sol_trx, reason='ContinueV02')
             get_measurements(result)
-            (founded, signature) = check_if_continue_returned(result)
+            (founded, signature_) = check_if_continue_returned(result)
             if founded:
-                return signature
-
+                signature = signature_
+                extra_sol_trx = True
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: This check will give incorrect result in some cases. In general Solana can mixes transaction for ALL iterations. In this case you can get CONTINUE, CONTINUE, ERROR, FINALIZE, CONTINUE, ERROR results.
I think you should print status of the transaction or error message.

File: proxy/plugin/solana_rest_api_tools.py [hunk 3]
```diff
@@ -828,7 +838,53 @@ def simulate_continue(signer, client, perm_accs, trx_accs, step_count):
...
+        logger.debug("COST %s %d %d %s %s %s %s %s",
+                     'None',
+                     cost,
+                     used_gas if used_gas else 0,
+                     'None',
+                     'None',
+                     sig,
+                     "extra" if extra_sol_trx else "ok",
+                     reason if reason else "None",
+                     )
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: IMHO: It is better to use only one `logger.debug(...)`. in this case, you can guarantee that the structure of the record will be the same (even if another programmer will correct this code in the future).

File: proxy/plugin/solana_rest_api_tools.py (hunk 2 above)
Comment (sinev-valentine) [reply proxy/plugin/solana_rest_api_tools.py]: fixed

File: proxy/plugin/solana_rest_api_tools.py (hunk 3 above)
Comment (sinev-valentine) [reply proxy/plugin/solana_rest_api_tools.py]: fixed

File: proxy/plugin/solana_rest_api_tools.py (hunk 1 above)
Comment (sinev-valentine) [reply proxy/plugin/solana_rest_api_tools.py]: fixed

**Review by s-medvedev — DISMISSED** [review]

File: proxy/plugin/solana_rest_api_tools.py [hunk 4]
```diff
@@ -558,19 +568,32 @@ def call_continue_bucked_0x0d(signer, client, perm_accs, trx_accs, steps, msg):
...
+
         for trx in result_list:
             confirm_transaction(client, trx)
             result = client.get_confirmed_transaction(trx)
+
+            extra_sol_trx = False
+            if result['result']['meta']['err']:
+                instruction_error =  result['result']['meta']['err']['InstructionError']
+                err = instruction_error[1]
+                if isinstance(err, dict)  and 'Custom' in err:
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: You should check that `Custom error` has id 1.

Comment (sinev-valentine) [reply proxy/plugin/solana_rest_api_tools.py]: fixed

File: proxy/plugin/solana_rest_api_tools.py [hunk 5]
```diff
@@ -490,7 +490,7 @@ def call_continue_bucked(signer, client, perm_accs, trx_accs, steps, eth_trx):
             if result['result']['meta']['err']:
                 instruction_error =  result['result']['meta']['err']['InstructionError']
                 err = instruction_error[1]
-                if isinstance(err, dict)  and 'Custom' in err:
+                if isinstance(err, dict)  and err['Custom'] == 1:
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: What happens if the `Custom` key is missed in the dictionary?
I suggest use `err.get('Custom', 0)` instead.
Did you get non-zero extra_sol_trx on tests?

Comment (sinev-valentine) [reply proxy/plugin/solana_rest_api_tools.py]: fixed

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**